compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -465,6 +465,40 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             );
         }
 
+        // Collect all possible candidates in the chain with same 'item_name'
+        let mut all_methods: Vec<super::probe::Candidate<'_>> = vec![];
+        if let Mode::MethodCall = mode && let SelfSource::MethodCall(mut rcvr_expr) = source {
+        while let hir::ExprKind::MethodCall(_path_segment, parent_expr, _args, _span) =
```

Comment:
You will likely want to keep around wither the span or the path_segment.span for the final error, so you can point at the right place in the user's expression when talking about the method being available, because there might be more than one chain in the method call that have the method available.